internal/gamepaddb/gamepaddb.go

```diff
@@ -228,14 +228,30 @@ func parseMappingElement(str string) (*mapping, error) {
 
 		if str[0] == '+' {
 			numstr = str[2:]
+			// Only use the positive half, i.e. 0..1.
 			min = 0
 		} else if str[0] == '-' {
 			numstr = str[2:]
-			max = 0
-		}
-
+			// Only use the negative half, i.e. -1..0,
+			// but invert the sense so 0 does not "press" buttons.
+			//
+			// In other words, this is the same as '+' but with the input axis
+			// value reversed.
+			//
+			// See SDL's source:
+			// https://github.com/libsdl-org/SDL/blob/f398d8a42422c049d77c744658f1cd2bb011ed4a/src/joystick/SDL_gamecontroller.c#L960
+			min, max = 0, min
+		}
+
+		// Map min..max to -1..+1.
+		//
+		// See SDL's source:
+		// https://github.com/libsdl-org/SDL/blob/f398d8a42422c049d77c744658f1cd2bb011ed4a/src/joystick/SDL_gamecontroller.c#L276
+		// then simplify assuming output range -1..+1.
+		//
+		// Yields:
 		scale := 2 / (max - min)
```

Comment:
Is it OK for scale to be negative?

Reply:
Yes, it is; note that tilda also causes this. In fact, my fix is kinda equivalent to keeping the code as is and just adding "tilda = !tilda" to the "-" case, but that'd be dirty and matches neither GLFW's nor SDL's implementation.

Reply:
OK so let's revisit when GLFW is updated.

Reply:
Out of curiosity, is it possible to use `~` and `-` at the same time?

Reply:
Absolutely; both in SDL and Ebitengine this then inverts the output signal after all other processing. The implementation of this is correct.

However no current gamecontroller mapping uses this combination.

There are also a few extra peculiar entries in gamecontrollerdb that synthesize an axis from two buttons like this:

`+rightx:b9,+righty:b3,-rightx:b4,-righty:b8`

This seems to indicate a gamepad with multiple D-pads, where the only way to achieve standard mapping is a combination like this. Ebitengine and GLFW both don't support this, and at least Ebitengine will simply ignore these mappings. It seems like this applies primarily to N64 controllers.

Reply:
I see. We don't plan to support such extra dpad buttons since the current standard layout cannot have them. If you think we have to fix this, please file this...

Reply:
Yeah; we maybe should ignore mapping to unknown buttons/axes though. Right now Ebitengine considers them unknown axes with names like "-leftx", so the definition still parses and the N64 controller would be usable apart from these four buttons (meaning it'd basically still be a NES-ish controller with an extra stick). I guess that's fine.